tools/rosbag/src/record.cpp

```diff
@@ -145,8 +145,12 @@ rosbag::RecorderOptions parseOptions(int argc, char** argv) {
         if (freq == 0)
         {
             opts.custom_record_freq.emplace(it->first.as<std::string>(), ros::Duration(-1));
-        } 
-        else 
+        }
+        else if (freq == -1)
+        {
+            opts.custom_record_freq.emplace(it->first.as<std::string>(), ros::Duration(0));
```

Comment:
now that this is `Duration(0)`, the condition below actually doesn't work:
```        
        else if(rectime > it->second)
        {
             it->second += interval * (1 + int((rectime.toSec() - it->second.toSec()) / interval.toSec()));
        }
```
should be 
```        
        else if(rectime > it->second)
        {
             if (interval > ros::Duration(0)) {
                 it->second += interval * (1 + int((rectime.toSec() - it->second.toSec()) / interval.toSec()));
             }
        }
```